Gen8/bdsp.py

```diff
@@ -202,3 +202,208 @@ def encounters():
             if i != len(map_names) - 1:
                 f.write("\n")
+
+def underground():
+    UG_ENCOUNT_02 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_02.json"
+    UG_ENCOUNT_03 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_03.json"
+    UG_ENCOUNT_04 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_04.json"
+    UG_ENCOUNT_05 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_05.json"
+    UG_ENCOUNT_06 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_06.json"
+    UG_ENCOUNT_07 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_07.json"
+    UG_ENCOUNT_08 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_08.json"
+    UG_ENCOUNT_09 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_09.json"
+    UG_ENCOUNT_10 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_10.json"
+    UG_ENCOUNT_11 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_11.json"
+    UG_ENCOUNT_12 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_12.json"
+    UG_ENCOUNT_20 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_20.json"
+    UG_SPECIAL_POKEMON = f"{SCRIPT_FOLDER}/bdsp/UgSpecialPokemon.json"
+    UG_RAND_MARK = f"{SCRIPT_FOLDER}/bdsp/UgRandMark.json"
+    TAMAGO_WAZA_TABLE = f"{SCRIPT_FOLDER}/bdsp/TamagoWazaTable.json"
+    TAMAGO_WAZA_IGNORE_TABLE = f"{SCRIPT_FOLDER}/bdsp/UgTamagoWazaIgnoreTable.json"
+    UG_POKEMON_DATA = f"{SCRIPT_FOLDER}/bdsp/UgPokemonData.json"
+
+    with open(UG_ENCOUNT_02, "r") as f:
+        ug_encount_02 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_03, "r") as f:
+        ug_encount_03 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_04, "r") as f:
+        ug_encount_04 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_05, "r") as f:
+        ug_encount_05 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_06, "r") as f:
+        ug_encount_06 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_07, "r") as f:
+        ug_encount_07 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_08, "r") as f:
+        ug_encount_08 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_09, "r") as f:
+        ug_encount_09 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_10, "r") as f:
+        ug_encount_10 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_11, "r") as f:
+        ug_encount_11 = json.load(f)["table"]
+
+    with open(UG_ENCOUNT_12, "r") as f:
+        ug_encount_12 = json.load(f)["table"]
+
+    with open(UG_SPECIAL_POKEMON, "r") as f:
+        ug_special_encounters = json.load(f)["Sheet1"]
+
+    with open(UG_RAND_MARK, "r") as f:
+        ug_rand_mark_json = json.load(f)["table"]
+
+    with open(TAMAGO_WAZA_TABLE, "r") as f:
+        tamago_waza_table_json = json.load(f)["Data"]
+
+    with open(TAMAGO_WAZA_IGNORE_TABLE, "r") as f:
+        tamago_waza_ignore_table_json = json.load(f)["Sheet1"]
+
+    with open(UG_POKEMON_DATA, "r") as f:
+        ug_pokemon_data_json = json.load(f)["table"]
+
+    ug_encount = bytes()
+    ug_special_pokemon = bytes()
+    ug_rand_mark = bytes()
+    tamago_waza_table = bytes()
+    tamago_waza_ignore_table = bytes()
+    ug_pokemon_data = bytes()
+
+    for pokemon in ug_encount_02:
+        ug_encount += b"\x02"
```

Comment:
General question for all of the ug_encount_XX loops:
1. Is this value even necessary? What do you use it for? Can we not assume fixed offsets assuming it is some sort of clarifying value to know where we are in the data?
2. Assuming it is necessary, can it be moved out of the loop? It gets duplicated for every pokemon entry

Reply:
The amount of pokemon in each table isn't the same so the other way around would be to replace this value with the amount of entries so we know how many are in each and then can run based on indices from there. The reason it is there for each is to identify which entry belongs to which table in the long run. 

The number of entries is fixed per but that would mean hard coding the number for each section within finder (which personally I'm not a fan of hard coded values like that).